bin/publish.ml

```diff
@@ -43,9 +66,9 @@ let publish_alt ~dry_run pkg kind =
   Pkg.publish_msg pkg >>= fun msg ->
   Delegate.publish_alt ~dry_run pkg ~kind ~msg ~archive
 
-let publish ?build_dir ?opam ?delegate ?change_log ?distrib_uri ?distrib_file
-    ?publish_msg ~name ~pkg_names ~version ~tag ~keep_v ~dry_run
-    ~publish_artefacts ~yes () =
+let publish ?(from_cli = false) ?build_dir ?opam ?delegate ?change_log
```

Comment:
This `from_cli` argument seems a bit convoluted to me. I think we could determine whether it was explicitly asked for with the `publish_artifact` argument, ie if it contains `Doc` then it was asked for explicitly. I think this makes things clearer.

I wonder if it wouldn't be worth changing `Pkg.publish_artifact` to `Pkg.default_publish_artifacts` and either use the cli args if they're non empty or the default ones. It would make things much clearer than having to populate the package with the args and always go through `Pkg`. I find it rather confusing.

Reply:
That's a good catch, I changed this.